src/components/SideNav/index.js

```diff
@@ -0,0 +1,32 @@
+import { useState } from 'react';
+import { NavLink } from 'react-router-dom';
+import KeyboardDoubleArrowLeftIcon from '@mui/icons-material/KeyboardDoubleArrowLeft';
+import KeyboardDoubleArrowRightIcon from '@mui/icons-material/KeyboardDoubleArrowRight';
+import cn from 'classnames';
+
+import { navData } from './navData';
+import './sidenav.css';
+
+const SideNav = () => {
+  const [open, setopen] = useState(false);
+
+  const toggleOpen = () => {
+    setopen(!open);
```

Comment:
```suggestion
    setOpen(prev => !prev);
```
I think this is the most correct way of doing it